### ISSUE TITLE
PiVPN stopped working, was working previously

### ISSUE THREAD

**Issue by Clubguy** [pivpn/pivpn#297]
# PiVPN Issue Template

## Console output of `curl install.pivpn.io | bash`
```

```

## Console output of `pivpn add` or `pivpn add nopass`
```

```

## Console output of `pivpn debug`
```
:::                                     :::
::              PiVPN Debug              ::
:::                                     :::
::      Latest Commit                    ::
:::                                     :::
commit [redacted]
Author: 0-kaladin <0.kaladin@gmail.com>
Date:   Mon Feb 27 21:01:23 2017 -0500

    Update README for IRC channel

    Due to discontinuation of Google Spaces
:::                                     :::
::      Recursive list of files in       ::
::      /etc/openvpn/easy-rsa/pki        ::
:::                                     :::
/etc/openvpn/easy-rsa/pki/:
[Redacted].ovpn
ca.crt
Default.txt
dh2048.pem
[Redacted].ovpn
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
[Redacted].ovpn
issued
private
serial
serial.old
ta.key

/etc/openvpn/easy-rsa/pki/issued:
[Redacted].crt
[Redacted].crt

/etc/openvpn/easy-rsa/pki/private:
[Redacted].key
ca.key
[Redacted].key

:::                                     :::
::      Output of /etc/pivpn/*           ::
:::                                     :::
:: START /etc/pivpn/DET_PLATFORM ::
Raspbian
:: END /etc/pivpn/DET_PLATFORM ::
:: START /etc/pivpn/INSTALL_PORT ::
443
:: END /etc/pivpn/INSTALL_PORT ::
:: START /etc/pivpn/INSTALL_PROTO ::
tcp
:: END /etc/pivpn/INSTALL_PROTO ::
:: START /etc/pivpn/INSTALL_USER ::
[Redacted]
:: END /etc/pivpn/INSTALL_USER ::
:: START /etc/pivpn/NO_UFW ::
1
:: END /etc/pivpn/NO_UFW ::
:: START /etc/pivpn/pivpnINTERFACE ::
eth0
:: END /etc/pivpn/pivpnINTERFACE ::
:: START /etc/pivpn/REVOKE_STATUS ::
0
:: END /etc/pivpn/REVOKE_STATUS ::
:::                                     :::
:: /etc/openvpn/easy-rsa/pki/Default.txt ::
:::                                     :::
client
dev tun
proto tcp
remote [Redacted] 443
resolv-retry infinite
nobind
persist-key
persist-tun
key-direction 1
remote-cert-tls server
tls-version-min 1.2
verify-x509-name server name
cipher AES-256-CBC
auth SHA256
comp-lzo
verb 1
:::                                     :::
::      Debug Output Complete            ::
:::                                     :::
```

## Issue
PiVPN was working great until a couple of days ago.  I was able to connect with up to 5 different devices, now none of them connect.  The service is running on the raspberry pi, but no luck connecting.

**Comment by cfcolaco** [comment]
the info you are providing is too vague, I will assume you are running raspbian Jessie, 

1 - Have you made any changes in your pi recently? ( updates, moved any files, edited config files, deleted / added users )  
2 - Can you provide the error you get when trying to connect? 
3 -Can you provide the output of `sudo iptables -S` and `sudo iptables -t nat -S`

**Comment by Clubguy** [comment]
The pi is also running Pi-hole but they have been working fine.  Pi-hole updates periodically, but it hasn't been a problem so far.

Error Message
Sun Jul 02 08:23:09 2017 TCP/UDP: Preserving recently used remote address: [AF_INET][Redacted]:443
Sun Jul 02 08:23:09 2017 Attempting to establish TCP connection with [AF_INET][Redacted]:443 [nonblock]
Sun Jul 02 08:25:09 2017 TCP: connect to [AF_INET][Redacted]:443 failed: Connection timed out (WSAETIMEDOUT)
Sun Jul 02 08:25:09 2017 SIGUSR1[connection failed(soft),init_instance] received, process restarting
Sun Jul 02 08:25:14 2017 TCP/UDP: Preserving recently used remote address: [AF_INET][Redacted]:443
Sun Jul 02 08:25:14 2017 Attempting to establish TCP connection with [AF_INET][Redacted]:443 [nonblock]

root@xxx:/tmp# sudo iptables -S
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT

root@xxx:/tmp# sudo iptables -t nat -S
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A POSTROUTING -s 10.8.0.0/24 -o eth0 -j MASQUERADE

**Comment by cfcolaco** [comment]
The error you are experiencing happens mostly because clients can't reach the server. 
Server Down // Firewall Blocking due to ports not beeing properly forwarded ... things like this. 

can you confirm that you have port 443 properly forwarded on your router? 

if port forwarding is not the issue please paste here the content of `/etc/openvpn/server.conf` and the content of one .ovpn file.

don't forget to hide any sensitive information, like public IPs and Keys.

**Comment by Clubguy** [comment]
Server.Conf
dev tun
proto tcp
port 443
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/server.crt
key /etc/openvpn/easy-rsa/pki/private/server.key
dh /etc/openvpn/easy-rsa/pki/dh2048.pem
topology subnet
server 10.8.0.0 255.255.255.0
# server and remote endpoints
ifconfig 10.8.0.1 10.8.0.2
# Add route to Client routing table for the OpenVPN Server
push "route 10.8.0.1 255.255.255.255"
# Add route to Client routing table for the OPenVPN Subnet
push "route 10.8.0.0 255.255.255.0"
# your local subnet
push "route [redacted] 255.255.255.0"
# Set your primary domain name server address for clients
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
duplicate-cn
keepalive 10 120
tls-version-min 1.2
tls-auth /etc/openvpn/easy-rsa/pki/ta.key 0
cipher AES-256-CBC
auth SHA256
comp-lzo
user nobody
group nogroup
persist-key
persist-tun
#crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
log /var/log/openvpn.log
verb 1
# Generated for use by PiVPN.io

OVPN File
client
dev tun
proto tcp
remote [redacted] 443
resolv-retry infinite
nobind
persist-key
persist-tun
key-direction 1
remote-cert-tls server
tls-version-min 1.2
verify-x509-name server name
cipher AES-256-CBC
auth SHA256
comp-lzo
verb 1
<ca>
-----BEGIN CERTIFICATE-----
[redacted]
-----END CERTIFICATE-----
</ca>
<cert>
-----BEGIN CERTIFICATE-----
[redacted]
-----END CERTIFICATE-----
</cert>
<key>
-----BEGIN ENCRYPTED PRIVATE KEY-----
[redacted]
-----END ENCRYPTED PRIVATE KEY-----
</key>
<tls-auth>
#
# 2048 bit OpenVPN static key
#
-----BEGIN OpenVPN Static key V1-----
[redacted]
-----END OpenVPN Static key V1-----
</tls-auth>

**Comment by Clubguy** [comment]
Bueller?

**Comment by Clubguy** [comment]
Should I wipe clean and start over?

**Comment by Clubguy** [comment]
Crickets

**Comment by cfcolaco** [comment]
sorry @Clubguy ... right now its impossible for me to help you since i am  on holidays and after i will be moving from my home country, if no one comes to help you meanwhile, i will come back as  soon as i can

**Comment by Clubguy** [comment]
No problem Carlos…where are you moving from and where are you moving to?

 

Do you think I should try to wipe the drive clean and start over or should I wait for you to settle in at your new place.

 

John

**Comment by cfcolaco** [comment]
@Clubguy 

sorry, I am moving from Portugal to Sweden. 

anyway, it is up to you if you want to start over or wait for me to have time and look for all the issue again, rethink about it and eventually will have something more to tell you. 

meanwhile, there's so much going on that I do not have much time. 
I pass by when I have 5 mins or so try to help with "smaller" issues and I go away! 

In August will have a bit more time.

**Comment by jsfrederick** [comment]
I,too, am having this problem.  PIVPN was working fine, and then just stopped.  I changed nothing, except maybe doing an apt-get update and apt-get upgrade on Raspbin Jessie.  I ended up reinstalling from scratch, reimagine Raspbian and reinstalling PIVPN. I have port 1194 forward to the PIVPN server.  Basically, when I try to connect from my iPad, I get at connection timeout.

  My server.conf is:

``
dev tun
proto udp
port 1194
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/server.crt
key /etc/openvpn/easy-rsa/pki/private/server.key
dh /etc/openvpn/easy-rsa/pki/dh2048.pem
topology subnet
server 10.8.0.0 255.255.255.0
# server and remote endpoints
ifconfig 10.8.0.1 10.8.0.2
# Add route to Client routing table for the OpenVPN Server
push "route 10.8.0.1 255.255.255.255"
# Add route to Client routing table for the OPenVPN Subnet
push "route 10.8.0.0 255.255.255.0"
# your local subnet
push "route 192.168.0.0 255.255.255.0"
# Set your primary domain name server address for clients
push "dhcp-option DNS 208.67.222.222"
push "dhcp-option DNS 208.67.220.220"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
duplicate-cn
keepalive 10 120
tls-version-min 1.2
tls-auth /etc/openvpn/easy-rsa/pki/ta.key 0
cipher AES-256-CBC
auth SHA256
comp-lzo
user nobody
group nogroup
persist-key
persist-tun
#crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
log /var/log/openvpn.log
verb 1
# Generated for use by PiVPN.io
``

**Comment by S10t** [comment]
I had the same problem. Which is the reason i found this thread.
changed nothing but suddenly could not connect to my pivpn.

As usual i started looking for far too complex issues. in the end i checked my WAN ip. Which should be static, but for no apparant reason changed. So a simple change in the client config to the right ip, solved all my problems.

**Comment by Clubguy** [comment]
When you say WAN, do you mean your IP from your internet provider changed?

**Comment by S10t** [comment]
With WAN i mean the external ip adress the internet provider gave you.
Check http://www.whatsmyip.org/
The ip adress stated there should be corresponding with you config files.

**Comment by Clubguy** [comment]
Carlos,

 

Have you had a chance to look at this since your move?

 

Thank you,

 

John

**Comment by cfcolaco** [comment]
@Clubguy you still facing this issue??  I had not yet the time to look to everything I want ... but things are getting more in line! ... I have been traveling since jully ... so ... being kind of hard ... but now I already have a traveler pi testing kit!! =p so when I get a break from my daily work ... I am really sorry for bot being able to help that much!

**Comment by cfcolaco** [comment]
@Clubguy anyway .. right know as many things have changed ... I would suggest you ... to do a fresh install and try it with raspbian stretch instead! then give me your feedback please!

**Comment by cfcolaco** [comment]
@jsfrederick  same for you!

**Comment by jsfrederick** [comment]
Thanks Carlos!  I just saw these emails come in, this is good news indeed!

I was just wondering this morning if I should try again as things have been updated all around.

I am also busy with work, so I will not be able to try for a few weeks.  Will follow up when I can.

Thanks!

**Comment by cfcolaco** [comment]
@jsfrederick yup .. the few time I had .. was just to help in issues that can be easily solved and to add the support for stretch .. so .. i think this should solve your issues!

**Comment by aditya12anand** [comment]
The problem with me was the same the PiVPN worked amazingly for three days but after that, when I booted it back on it stopped working.

i) The port forwarding on my router has been configured properly (hence it worked the first time)
ii) The issue is that is being displayed is that the "Waiting for server response" (I pinged my raspberry pi from my device and the connection was up)
iii) I checked the openvpn status on my raspberry pi and it was active 
iv) The client .opvn files have been correctly added to the tunnelblick properly
v) I checked wireshark for more information and it showed [Malformed Packet] every connection that was taking plave on that PORT
vi) The TLS hanshake is failing, this is being showed in the debugging report by Tunnelblick.

Now can someone please tell me what else should I be looking for? Cause these are the basic check that I thought I must carry out and now I have no idea what else could be wrong.

**Comment by NathanielJS1541** [comment]
I am currently having the same issue too. Since I have given a few of my friends config files for us to use at college (Since the security certificates have not been properly set up on the collage Internet, which doesn't seem to secure), I am reluctant to just wipe it and start again, as this would mean having to create each of the config files and distribute them again.

**Comment by were491** [comment]
(only use this post if pivpn stopped working after a reboot of the service or the pi, otherwise you are wasting your time and I don't want that)



Old AF thread but this happens so often and I always solved it with this, don't knoe about you:

First, uninstall Pi-Hole (you can reconfig it later, but you might want to keep your old settings in /etc/openvpn).

Then, 

install easy-rsa with

sudo apt-get install easy-rsa

After it's done, do:

sudo mkdir /etc/openvpn

sudo mkdir /etc/openvpn/easy-rsa

sudo cp -R /usr/share/easy-rsa/* /etc/openvpn/easy-rsa 

Setup pi-hole as normal.

I did this a while ago, so I might have forgotten some steps, but this is what I remember worked for me.
source:
https://www.digitalocean.com/community/questions/no-easy-rsa-to-install-on-my-debian

raspberrypi.stackexchange.com/questions/66961/installing-easy-rsa-examples-onto-a-raspberry-pi-3

**Comment by halpingregory** [comment]
Well I found the solution for me - hopefully it helps you too!

Found that /var/log/openvpn.log on the RPi had the line "VERIFY ERROR: depth=0, error=CRL has expired" Searching that gave several results saying to regenerate the Certificate Revocation List, but I couldn't find how to do this with pivpn.

Found this workaround and it worked: use pivpn to add a new user, then revoke the user - the revoke output notes that it regenerates the CRL. I was then able to connect again straightaway :) https://forums.openvpn.net/viewtopic.php?t=26585
